### DISCUSSION TITLE
Idiot’s guide to compiling grblHAL for RP2040 in Windows.

### DISCUSSION THREAD

**Discussion by cstaton-ftlg** [grblHAL/RP2040#82] [General]
Hi, my name is Chuck and I'm an idiot. I spent 15 straight hours yesterday trying to figure out how to compile grblHAL for my BTT SKR Pico board (didn't use web builder because I needed to modify it) and when I finally figured it out, I spent another 3 hours going through the whole process again on a fresh install of Windows so that I could document the process with lots of pretty pictures. I leave it here for the benefit of my fellow idiots.

[Idiot’s guide to compiling grblHAL for RP2040 in Windows..pdf](https://github.com/grblHAL/RP2040/files/13846709/Idiot.s.guide.to.compiling.grblHAL.for.RP2040.in.Windows.pdf)

Please forgive me if anything in there is not "the proper way" to go about things, but it does work.

**Comment by techsahli** [comment]
thanks for your effort and time spent

**Comment by ObiFredKenobi** [comment]
Just wanted to say thank you @cstaton-ftlg your guide really helped!

**Comment by cstaton-ftlg** [comment]
@techsahli & @ObiFredKenobi I found the grblHAL project itself hugely helpful and I am happy to see that the guide helped someone else utilize grblHAL. Honestly I thought nobody would read it, or if they did, would tell me I did it all wrong.
